src/test/kotlin/org/rust/ide/actions/RsFunctionMoverTest.kt

```diff
@@ -0,0 +1,422 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.actions
+
+class RsFunctionMoverTest : RsBaseMoverTest() {
+
+    fun `test step selection`() = moveBothDirectionTest("""
+            // - main.rs
+            fn <selection>test() {</selection>
+
+            }
+
+        """, """
+            // - main.rs
+
+            fn <selection>test() {</selection>
+
+            }
+        """)
+
+    fun `test step`() = moveBothDirectionTest("""
+            // - main.rs
+            fn /*caret*/test() {
+
+            }
+
+        """, """
+            // - main.rs
+
+            fn /*caret*/test() {
+
+            }
+        """)
+
+    fun `test step over function multi liner`() = moveBothDirectionTest("""
+            // - main.rs
+            fn /*caret*/test() {
+
+            }
+            fn foo() {
+
+            }
+        """, """
+            // - main.rs
+            fn foo() {
+
+            }
+            fn /*caret*/test() {
+
+            }
+        """)
+
+    // This test fails in run intellij
```

Comment:
No, of the top of my head I don't know what happens here. So looks like the best way to debug this is to launch IDEA under debugger and just step through. 

However, the behavior in IntelliJ seems more reasonable: if there's a single blank line between two functions, we should swap them. Kotlin preserves all the whitespace between functions as well, so perhaps we should do the same?